src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,609 @@
+
+DATA_DIR = Path(os.environ["AA_DATA_DIR"])
+PUBLIC_DATA_DIR = "public"
+RAW_DATA_DIR = "raw"
+PROCESSED_DATA_DIR = "processed"
+ARC2_DIR = "arc2"
+
+# TODO: Should add in parameters that control whether
+# or not you want to save the intermediate / raw data
+
+# TODO: Parameters should also control whether or not a function,
+# like computing dry spells, uses pre downloaded data or whether
+# it should download the data in one go
+
+# TODO: Error handling for various things,
+# like if data is missing or certain files don't exist
+# Need to do better exceptions (not bare or default)
+
+# TODO: Better logging
+
+# TODO: Write tests
+
+
+class ARC2:
+    """Summary of class
+
+    TODO: fill
+
+    Attributes:
+        country_iso3: ISO3 string.
+        date_min: Minimum date to load data from, either string
+            in ISO 8601 format, e.g. '2021-03-20' or `datetime.date` object.
+        date_max: Maximum date to load data from, either string
+            in ISO 8601 format, e.g. '2021-04-20' or `datetime.date` object.
+        range_x: Tuple of strings specifying longitude range
+            for download, e.g. ('32E', '36E')
+        range_y: Tuple of strings specifying latitude range
+            for download, e.g. ('20S', '5S')
+    """
+
+    def __init__(
+        self,
+        country_iso3: str,
+        date_min: Union[str, date],
+        date_max: Union[str, date],
+        range_x: Tuple[str, str],
+        range_y: Tuple[str, str],
+    ):
+        self.country_iso3 = country_iso3
+
+        if not isinstance(date_min, date):
+            date_min = date.fromisoformat(date_min)
+        self.date_min = date_min
+
+        if not isinstance(date_max, date):
+            date_max = date.fromisoformat(date_max)
+        self.date_max = date_max
+
+        self.range_x = range_x
+        self.range_y = range_y
+
+    def _download(
+        self, date_min_dl: date, date_max_dl: date, master: bool = False
+    ):
+
+        """
+        Download data from IRI servers for a specific date range
+        and longitude/latitude bounds. If `master`, the file
+        is fully loaded as the master file. Otherwise, it is
+        temporarily loaded and merged into the master file before
+        removal.
+
+        :param date_min_dl: `datetime.date` specifying first date
+            to download from.
+        :param date_max_dl: `datetime.date` specifying last date
+            to download from.
+        :param master: Boolean specifying whether download is master
+            file. If `True`, existing master is removed and full
+            file is saved. If `False`, download is temporarily saved
+            and then merged to master and deleted.
+        """
+
+        url = (
+            f"https://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NCEP/.CPC/.FEWS/"
+            f".Africa/.DAILY/.ARC2/.daily/.est_prcp/T/"
+            f"%28{date_min_dl:%-d%%20%b%%20%Y}%29"
+            f"%28{date_max_dl:%-d%%20%b%%20%Y}%29RANGEEDGES/"
```

Comment:
Oooh interesting! Good catch (and sorry that was left to you!). Have replaced `%-d` with `date_min_dl.day` and should now be a cross platform solution since that was the only platform dependent formatter there. Just pushed solution if you want to test.